### ISSUE TITLE
The execution of a WASM transaction might lead to an un-deterministic node state

### ISSUE THREAD

**Issue by dzmitryhil** [CosmWasm/wasmd#1688]
The execution of a WASM transaction might lead to an un-deterministic node state when one validator rejects the transaction with error: 
```
failed to execute message; message index: 0: Error calling the VM: Error instantiating a Wasm module: Error instantiating module: CpuFeature("EnumSet(AVX512DQ | AVX512VL | AVX512F)"): execute wasm contract failed
  ```
As a result, the validator node is halted because of the app state hash diff.

Reproduced with: 
`github.com/CosmWasm/wasmd v0.30.0`
`github.com/CosmWasm/wasmvm v1.1.2`

**Comment by alpe** [comment]
> There are two common cases how this happened in the past:
> * User changes CPU of a machine but keeps the cache directory
>* User uses snapshot which contains the cache folder
>
>Those issues are solved in wasmvm 1.3 which includes the CPU type in the cache folder, invalidating caches from other CPUs.
Deleting the cache folder should solve the problem in the meantime.

Thanks to @webmaster128 and @chipshort for this

**Comment by alpe** [comment]
Thanks for reporting!
I will close the issue now but feel free to reopen and comment if this is not solving your problem
